### ISSUE TITLE
Help debugging issue with rubocop linter

### ISSUE THREAD

**Issue by patrykk21** [rubyide/vscode-ruby#441]
### Your environment

- `vscode-ruby` version: rebornix.ruby-0.22.2
- Ruby version: ruby 2.3.4p301 (2017-03-30 revision 58214) [x86_64-darwin18], however I switch based on project
- Ruby version manager (if any): RVM
- VS Code version: Version 1.31.1 (1.31.1)
- Operating System: OSX Mojave 10.14.3
- Using language server? I'm unsure on what this is. I know there's a setting in the settings.json for it. I have the issue both setting it to true and false

### Expected behavior

It lints files by `rubocop`.

### Actual behavior

Doesn't lint files by `rubocop` however it seems working the launching `rubocop` through terminal.

So I'm quite unsure on what I'm doing wrong and by looking on other issues and google I just couldn't figure out what I'm missing.

```json
  "ruby.codeCompletion": "rcodetools",
  "ruby.intellisense": "rubyLocate",
  "ruby.lint": {
    "debride": false,
    "fasterer": false,
    "reek": false,
    "rubocop": {
      "lint": true,
      "rails": true
    }
  },
  "ruby.useLanguageServer": false,
  "ruby.useBundler": false
```

```console
▶ gem env gemdir
/Users/patrykkotarski/.rvm/gems/ruby-2.3.4

▶ gem list | grep rubocop
rubocop (0.53.0)

▶ which rubocop
/Users/patrykkotarski/.rvm/gems/ruby-2.3.4/bin/rubocop

▶ rubocop -v
0.53.0
```

I get nothing in the `Problems` tab on vs-code apart from the Ruby linter error here below.

Ruby linter works as
```
Lint/AssignmentInCondition: Use `==` if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition.(undefined)
```

I got this error on developer tools:

```
INFO no standard startup: not a new window
log.ts:173   ERR Cannot read property 'then' of undefined: TypeError: Cannot read property 'then' of undefined
	at Locate.listInFile (/usr/local/share/.vscode/extensions/rebornix.ruby-0.22.2/out/locate/locate.js:69:29)
	at Object.provideDocumentSymbols (/usr/local/share/.vscode/extensions/rebornix.ruby-0.22.2/out/providers/intellisense.js:55:31)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:536:745
	at t.asPromise (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:50:385)
	at new Promise (<anonymous>)
	at Object.t.asPromise (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:50:353)
	at e.provideDocumentSymbols (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:536:705)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:554:222
	at e._withAdapter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:553:292)
	at e.$provideDocumentSymbols (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:554:184)
	at t._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:644:118)
	at t._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:643:744)
	at t._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:642:256)
	at t._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:641:155)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:638:960
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:38:994
	at e.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:44:113)
	at a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:170:213)
	at Socket.f._socketDataListener (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:170:434)
	at Socket.emit (events.js:182:13)
	at addChunk (_stream_readable.js:279:12)
	at readableAddChunk (_stream_readable.js:264:11)
	at Socket.Readable.push (_stream_readable.js:219:10)
	at Pipe.onread (net.js:636:20)
```

So I disabled `rubyLocate` and reloaded the window:

```
[Extension Host] [ruby] locate: walk completed! (1035msec)
console.ts:134 [Extension Host] Config changed. Should lint: 1
console.ts:134 [Extension Host] [ruby] locate: walk completed! (771msec)
log.ts:173   ERR An unknown error occurred. Please consult the log for more details.
```

Then I opened up the terminal of vs code and did
`gem install rubocop`
as of the [issue](https://github.com/rubyide/vscode-ruby/issues/109)

```
▶ gem install rubocop
Successfully installed rubocop-0.65.0
Parsing documentation for rubocop-0.65.0
Done installing documentation for rubocop after 6 seconds
1 gem installed
```

```
▶ rubocop -v
0.53.0
```

None of the related issues seems to be helping me.

Please, ask for any more info you'd need.

**Comment by wingrunr21** [comment]
If you are not using the language server then environment detection will not work and you will need to make sure VSCode can find the `rubocop` command. There are a number of ways to achieve this but the best way is to open VSCode from your project folder (via `code .`). This way VSCode inherits your shell environment and can find rubocop.

As for rubyLocate, you're the second person to report that error (see #430). I'm not sure what's going on there and haven't looked at that yet.

As for what the language server is, it's [in the README](https://github.com/rubyide/vscode-ruby#language-server)

**Comment by patrykk21** [comment]
@wingrunr21 wonderful! I didn't know about the Output tab. Then I also had to select 'Ruby Language Server' in the dropdown.

The command is running as expected
`executing rubocop -s <file_path> -f json -R -l`

It was returning no offences even though I intentionally changed code to make it weird and non ruby-style compliant.

At the end I found out the `-l` parameter or `"lint": "true"` in the `settings.json` was the issue. 

Disabling lint made it work for me.

Sorry for bothering and thank you!
